components/Dialogs/examples/DialogsThemedPresentationExampleViewController.swift

```diff
@@ -0,0 +1,202 @@
+                         toItem: self.view,
+                         attribute: .centerX,
+                         multiplier: 1.0,
+                         constant: 0.0),
+      ])
+  }
+
+  override var preferredContentSize: CGSize {
+    get {
+      return CGSize(width:200.0, height:150.0);
+    }
+    set {
+      super.preferredContentSize = newValue
+    }
+  }
+
+}
+
+class DialogsThemedPresentationExampleViewController: UIViewController {
+
+  private let materialButton = MDCButton()
+  private let transitionController = MDCDialogTransitionController()
+  var containerScheme = MDCContainerScheme()
+
+  override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: Bundle?) {
+    super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)
+    containerScheme.colorScheme = MDCSemanticColorScheme(defaults: .material201804);
+  }
+
+  required init?(coder aDecoder: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
+
+  override func viewDidLoad() {
+    super.viewDidLoad()
+
+    view.backgroundColor = containerScheme.colorScheme?.backgroundColor ??
+      MDCSemanticColorScheme(defaults: .material201804).backgroundColor
+
+    materialButton.translatesAutoresizingMaskIntoConstraints = false
+    materialButton.setTitle("Custom Alert With Themed Presentation", for: .normal)
+    materialButton.setTitleColor(UIColor(white: 0.1, alpha:1), for: .normal)
+    materialButton.sizeToFit()
+    materialButton.addTarget(self, action: #selector(tap), for: .touchUpInside)
+    materialButton.applyTextTheme(withScheme: containerScheme)
+    self.view.addSubview(materialButton)
+
+    NSLayoutConstraint.activate([
+      NSLayoutConstraint(item:materialButton,
+                         attribute: .centerX,
+                         relatedBy: .equal,
+                         toItem: self.view,
+                         attribute: .centerX,
+                         multiplier: 1.0,
+                         constant: 0.0),
+      NSLayoutConstraint(item: materialButton,
+                         attribute: .centerY,
+                         relatedBy: .equal,
+                         toItem: self.view,
+                         attribute: .centerY,
+                         multiplier: 1.0,
+                         constant: 0.0)
+      ])
+  }
+
+  @objc func tap(_ sender: Any) {
+    let title = "Reset Settings"
+    let message = "Your device will reset to its default factory settings."
+    let customDialogController = CustomDialogViewController(title: title, message: message)
+    customDialogController.modalPresentationStyle = .custom;
+    customDialogController.transitioningDelegate = self.transitionController;
+
+    if let presentationController = customDialogController.mdc_dialogPresentationController {
+      presentationController.applyTheme(withScheme: containerScheme)
+      presentationController.dialogCornerRadius = customDialogController.view.layer.cornerRadius
```

Comment:
Why are you changing the presentation controller's corner radius? It should be the other way around.

Reply:
It's a good comment.

I was suggested to check out the example here: https://github.com/material-components/material-components-ios/blob/develop/components/Dialogs/examples/DialogsCustomShadowExampleViewController.swift#L106, and followed the rules here.

Originally I thought it should the themer changes the cornerRadius and then the presentedController set the cornerRadius accordingly, but maybe the example posted above is just a special case?

Reply:
I changed it in 7c3b8a2b897c3552b19b33a3ae50a2c7d9c9ee77. What do you think?

Reply:
Present screenshot looks so much better! Thank you!